packages/graphql-server/src/functions/useRequireAuth.ts

```diff
@@ -0,0 +1,69 @@
+import type { APIGatewayEvent, Context as LambdaContext } from 'aws-lambda'
+
+import { getAuthenticationContext } from '@redwoodjs/api'
+
+import {
+  getAsyncStoreInstance,
+  setContext,
+  context as globalContext,
+} from '../globalContext'
+
+import type { GetCurrentUser } from './types'
+
+export const useRequireAuth = ({
+  handlerFn,
+  getCurrentUser,
+}: {
+  handlerFn: (
```

Comment:
I think maybe.... the handlerFn could also be async, not 100% sure but  Ithink TS will error if you give `useRequireAuth` an async handler

Reply:
My tests have the handler as an async:

```ts
const handler = async (
  _event: APIGatewayEvent,
  _context: Context
): Promise<any> => {
  // @MARK
  // Don't use globalContext until beforeAll runs
```

and then `handlerFn` returns a Promise:

```ts
export const useRequireAuth = ({
  handlerFn,
  getCurrentUser,
}: {
  handlerFn: (
    event: APIGatewayEvent,
    context: LambdaContext,
    ...others: any
  ) => any
  getCurrentUser: GetCurrentUser
}) => {
  return async (
    event: APIGatewayEvent,
    context: LambdaContext,
    ...rest: any
  ): Promise<any> => {
```

I think we are ok. If not can fix in next release since this is a more of a preview feature.